### PR TITLE
Add function `date_or_null`

### PR DESCRIPTION
**Pull request by joverlee521**
This function casts a string to a date or returns null if the string is in an invalid datetime format.

~This function does not protect against dates that are out of range! Do we want it to?~
This function also returns null if a date is out of range i.e. `2020/01/50`

### CODE REVIEW

**Comment by joverlee521** [comment]
[valid date function](https://trello.com/c/HmlXxzLY/319-valid-date-function)

**Comment by tsibley** [comment]
> This function does not protect against dates that are out of range! Do we want it to?

Nice catch! I think we do probably want it to protect against all bad input, so returning null given something like `2020-01-50`.

File: schema/verify/functions/date_or_null.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        begin
+            perform date_or_null('2020-01-50');
+            raise exception 'date_or_null() did not raise an error on out of range date';
+        exception
+            when datetime_field_overflow then
+                null; -- expected!
+        end;
+    end
+$$;
```
Comment (tsibley) [inline schema/verify/functions/date_or_null.sql]: Great to see these tests! I don't know about you, but I find it really satisfying to write tests of simple functions like this. :nerd_face:

Comment (joverlee521) [reply schema/verify/functions/date_or_null.sql]: Yes. _Very_ satisfying 🤓

**Pull request merged**